sling/nlp/parser/trainer/transition-generator.cc

```diff
@@ -0,0 +1,457 @@
+// Copyright 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <deque>
+#include <vector>
+
+#include "sling/frame/object.h"
+#include "sling/frame/store.h"
+#include "sling/nlp/parser/trainer/transition-generator.h" 
+
+namespace sling {
+namespace nlp {
+namespace {
+
+// Map from a frame to information pertinent to transition generation.
+class FrameInfo {
```

Comment:
Is it correct that "FrameInfo" is basically a frame graph where the nodes are Entry objects and edges are Edge objects? If so, you could consider using a terminology that is more graph-like, i.e. graphs, nodes, and edges.

Reply:
Done.